src/common/AdaptiveStream.cpp

```diff
@@ -1289,16 +1320,15 @@ bool AdaptiveStream::GenerateSidxSegments(PLAYLIST::CRepresentation* rep)
 
 void AdaptiveStream::Stop()
 {
-  if (current_rep_)
-  {
-    current_rep_->SetIsEnabled(false);
-  }
-
   if (thread_data_)
   {
     thread_data_->Stop();
     StopWorker(STOPPED);
   }
+  // Disable representation only after stopped the worker
+  // otherwise if read some segments may invalidate this change
+  if (current_rep_)
+    current_rep_->SetIsEnabled(false);
```

Comment:
Here the problem should be due to `AdaptiveStream::ensureSegment`
since force change `SetIsEnabled` on the representation and race condition may happens